js/viz/axes/datetime_breaks.js

```diff
@@ -0,0 +1,201 @@
+"use strict";
+
+var dateUtils = require("../../core/utils/date"),
+    days = [0, 1, 2, 3, 4, 5, 6];
+
+function getWeekendDays(workdays) {
+    return days.filter(function(day) {
+        return !workdays.some(function(workDay) {
+            return workDay === day;
+        });
+    }).map(function(day) {
+        return days.indexOf(day);
+    });
+}
+
+function getNextDayIndex(dayIndex) {
+    return (++dayIndex) % 7;
```

Comment:
Why you change value of dayIndex? It's equal to `(dayIndex + 1) % 7; ` expression.